### ISSUE TITLE
Don't pre-apprend baseurl to each fragment in DASH/other fragmented protocols

### ISSUE THREAD

**Issue by wiiaboo** [rg3/youtube-dl#12990]
- [X] I've **verified** and **I assure** that I'm running youtube-dl **2017.05.01**
- [X] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [X] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [X] Feature request (request for a new functionality)

---

Example URL: 
- https://www.youtube.com/watch?v=XmL19DOP_Ls (9h53m40s)

As it is now, with each fragment url being pre-appended by the baseURL from the manifest, the JSON exported from this URL reaches 31.1MB. If I change Youtube-DL to not do this, the resulting JSON is only 5.45MB.

Considering baseURL is already available in the ".requested_formats[*].url" key, the API user could check if the fragment URL starts with 'http' and pre-append the baseURL itself if not, or some other new key could be added for the baseURL.

**Comment by yan12125** [comment]
Related: #9109, which removes fragment URLs completely

**Comment by wiiaboo** [comment]
Fragment URLs are still needed for mpv until FFmpeg gets a DASH demuxer, though. It's not like the huge JSON prohibits playback, it's just a lot of repeated text for no reasonable gain.

**Comment by yan12125** [comment]
#9109 is a step towards live DASH streams. (#10787) As fragment URLs change as time goes by, it's not a good idea for youtube-dl to place them in extractor outputs (```-j```). Do you know how mpv handles live DASH streams?

**Comment by wiiaboo** [comment]
It doesn't, most likely. Any example URLs?

**Comment by yan12125** [comment]
Facebook live videos use DASH. A current example is
https://www.facebook.com/cs50/videos/10154276859382127/. You can find more
on https://www.facebook.com/livemap. Note that you'll need to login to get
video URLs on the livemap.

**Comment by wiiaboo** [comment]
That works fine though, that's not segmented dash.

**Comment by wiiaboo** [comment]
Youtube-DL doesn't do anything with it either. It just downloads the manifest.

**Comment by yan12125** [comment]
> Youtube-DL doesn't do anything with it either. It just downloads the manifest.

Live DASH streams are not supported yet, so I decide to just return the DASH URL so that users can feed it to capable players. For example:
```
MP4Client $(youtube-dl -g url_of_the_live_video)
```
I mention the case here because I hope there will be a good design so that both mpv and youtube-dl will work for live DASH in the future. Personally I don't like the ```youtube-dl -o - | mpv -``` way as it's not the same smooth as the youtube-dl integration in mpv.

**Comment by wiiaboo** [comment]
The future will be a working/valid DASH demuxer in FFmpeg. It's probably out-of-scope for mpv.

Trying the patch in the ML it seems to work at least with Youtube's and Vimeo's non-live .mpd.

**Comment by yan12125** [comment]
Once DASH support lands in ffmpeg, mpv doesn't need fragment URLs from youtube-dl anymore, right?

For now, minimizing ```-j``` output is still a good feature for static (non-live) MPDs.

**Comment by wiiaboo** [comment]
Since it's a huge patch with potentially a few security issues it won't be merged so soon (maybe months).
mpv would be fine with live DASH not working meanwhile, as far as my opinion goes.
